integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/BookResourceTest.java

```diff
@@ -0,0 +1,300 @@
+package io.quarkus.it.mongodb.panache;
+
+import static io.restassured.RestAssured.get;
+
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+import io.quarkus.it.mongodb.panache.book.BookDetail;
+import io.quarkus.it.mongodb.panache.person.Person;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import io.restassured.config.ObjectMapperConfig;
+import io.restassured.mapper.TypeRef;
+import io.restassured.parsing.Parser;
+import io.restassured.response.Response;
+
+@DisabledOnOs(OS.WINDOWS)
+@QuarkusTest
+@Testcontainers
+class BookResourceTest {
+    private static final TypeRef<List<BookDTO>> LIST_OF_BOOK_TYPE_REF = new TypeRef<List<BookDTO>>() {
+    };
+    private static final TypeRef<List<Person>> LIST_OF_PERSON_TYPE_REF = new TypeRef<List<Person>>() {
+    };
+
+    @Container
+    private static final GenericContainer MONGO = new FixedHostPortGenericContainer("mongo:4.0")
+            .withFixedExposedPort(27018, 27017);
```

Comment:
I use test containers because mongodb-embedded that was use in the mongodb-client extension causes issue on my laptop (need to run the test as root). Other extension would also want to use test containers because it's very conveniant (Vault extension), there is an issue opened about it.
IMHO leveraging test containers can be very conveniant for Quarkus integration tests (for example to test different DBs).
But now that I finish the implementation, I can switch back to embeded MongoDB as it is used by mongodb-client extension.

Reply:
I go back to flapddodle for embedded mongo to be able to run in Windows